dwave/plugins/torch/boltzmann_machine.py

```diff
@@ -87,6 +97,79 @@ def sufficient_statistics(
         """
 
+    def estimate_beta(self, spins: torch.Tensor) -> float:
+        """Estimate the maximum pseudolikelihood temperature using
+        ``dwave.system.temperature``.
+
+        Args:
+            spins (torch.Tensor): A tensor of shape (b, N) where b is the sample size,
+                and N denotes the number of variables in the model.
+
+        Returns:
+            float: The estimated effective inverse temperature of the model.
+        """
+        h, J = self.ising
+        bqm = BinaryQuadraticModel.from_ising(h, J)
+        beta = 1.0 / maximum_pseudolikelihood_temperature(bqm, spins.numpy())[0]
+        return beta
+
+    def _pad(self, x: torch.Tensor) -> torch.Tensor:
+        """Pads the observed spins with ``torch.nan``s at ``self.hidx`` to mark them as
+        hidden units.
+
+        Args:
+            x (torch.Tensor): Partially-observed spins of shape (b, N) where b is the
+                batch size and N is the number of visible units in the model.
+
+        Raises:
+            ValueError: Fully-visible models should not ``_pad`` data.
+
+        Returns:
+            torch.Tensor: A (b, N) tensor of spin variables where N is the total number
+                of variables, i.e., number of visible and hidden units.
+        """
+        if self.fully_visible:
+            raise ValueError("Fully-visible models should not `_pad` data.")
+        bs, n_vis = x.shape
+        n_hid = self.hidx.shape[0]
+        n = n_vis + n_hid
+        padded = torch.ones((bs, n)) * torch.nan
+        padded[:, self.vidx] = x
+        return padded
+
+    @abstractmethod
+    def _compute_effective_field(self) -> torch.Tensor:
+        """Compute the effective field of disconnected hidden units.
+
+        Returns:
+            torch.Tensor: effective field of hidden units
+        """
+
+    def _compute_expectation_disconnected(
+        self, obs: torch.Tensor, beta: float
+    ) -> torch.Tensor:
+        """Compute and return the conditional expectation of spins including observed
+        spins.
+
+        Args:
+            obs (torch.Tensor): A tensor of spins with shape (b, N) where b is the
+                sample size and N is the number of visible units in the model.
+
+            beta (float): The effective inverse temperature of the model and sampler.
+                This quantity is, in the typical context of using a D-Wave QPU,
+                estimated with samples from the QPU and
+                :meth:`AbstractBoltzmannMachine.estimate_beta`.
+
+        Returns:
+            torch.Tensor: A (b, N)-shaped tensor of expected spins conditioned on
+                ``obs`` where b is the sample size and N is the total number of
+                variables in the model, i.e., number of hidden and visible units.
+        """
+        m = self._pad(obs)
+        h_eff = self._compute_effective_field(m)
+        m[:, self.hidx] = -torch.tanh(h_eff * beta)
```

Comment:
shouldn't we sample hidden units using torch.bernoulli once we calculate their probabilities?